### PR TITLE
Version 2.1 Release 

### PR DESCRIPTION
**Pull request by teamwork523**
Changes from the beta v2.1:
1. Enable scheduled task over Wifi
2. Ipv6 aware ping program selection
3. Ignore ip_address field before submission to GAE
4. Check available measurement types during checkin
5. Fix TCP Throughput result JSON conversion problem

### CODE REVIEW

File: android/AndroidManifest.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
   <uses-sdk android:targetSdkVersion="9" android:minSdkVersion="9" />  
 
   <application
-    android:debuggable="false"
+    android:debuggable="true"
```
Comment (ghost) [inline android/AndroidManifest.xml]: probably want this to be false.

File: android/res/values/strings.xml [hunk 2]
```diff
@@ -50,4 +50,6 @@
     <string name="defaultUser">Anonymous</string>
     <string name="TCPThroughputDirLabel">Direction:</string>
     <string name="TCPThroughputTargetHint">m-lab</string>
+    <string name="ping6_executable">/system/bin/ping6</string>
```
Comment (ghost) [inline android/res/values/strings.xml]: can you move this next to the existing ping entry?

File: android/src/com/mobiperf/BatteryCapPowerManager.java [hunk 3]
```diff
@@ -127,10 +127,11 @@ public MeasurementResult call() throws MeasurementError {
           Logger.i("Skipping measurement - low battery");
           throw new MeasurementSkippedException("Not enough battery power");
         }
-        if (PhoneUtils.getPhoneUtils().getNetwork() == PhoneUtils.NETWORK_WIFI) {
+        // Enable scheduled task over Wifi
+        /*if (PhoneUtils.getPhoneUtils().getNetwork() == PhoneUtils.NETWORK_WIFI) {
```
Comment (ghost) [inline android/src/com/mobiperf/BatteryCapPowerManager.java]: you can delete this section instead of commenting it out. we have version control shuold we want to bring it back.

File: android/src/com/mobiperf/measurements/PingTask.java [hunk 4]
```diff
@@ -148,8 +149,11 @@ public MeasurementTask clone() {
   @Override
   public MeasurementResult call() throws MeasurementError {
     PingDesc desc = (PingDesc) measurementDesc;
+    boolean isIPv6;
     try {
       InetAddress addr = InetAddress.getByName(desc.target);
+      // check resolved address length is 4 bytes (ipv4) or 16 bytes (ipv6)
```
Comment (ghost) [inline android/src/com/mobiperf/measurements/PingTask.java]: the comment doesn't match the code. You don't check if it's 4 bytes or 16 bytes, you just set the boolean based on if it's 16 bytes.

you might want to do both.

File: android/src/com/mobiperf/measurements/TCPThroughputTask.java [hunk 5]
```diff
@@ -306,12 +289,24 @@ public static Class getDescClass() throws InvalidClassException {
 
   @Override
   public void stop() {
+    System.gc();
```
Comment (ghost) [inline android/src/com/mobiperf/measurements/TCPThroughputTask.java]: why is this necessary?

File: android/src/com/mobiperf/measurements/TracerouteTask.java [hunk 6]
```diff
@@ -176,7 +174,13 @@ public MeasurementResult call() throws MeasurementError {
     
     
     try {
-      hostIp = InetAddress.getByName(target).getHostAddress();
+      InetAddress hostInetAddr = InetAddress.getByName(target);
+      hostIp = hostInetAddr.getHostAddress();
+      // TODO(Wenjie): Add a exhaustive list of ping locations for different Android phones
+      // add support for ipv6
```
Comment (ghost) [inline android/src/com/mobiperf/measurements/TracerouteTask.java:179]: you can remove part of this TODO now you've added ping6

File: android/src/com/mobiperf/measurements/TracerouteTask.java [hunk 7]
```diff
@@ -176,7 +174,13 @@ public MeasurementResult call() throws MeasurementError {
     
     
     try {
-      hostIp = InetAddress.getByName(target).getHostAddress();
+      InetAddress hostInetAddr = InetAddress.getByName(target);
+      hostIp = hostInetAddr.getHostAddress();
+      // TODO(Wenjie): Add a exhaustive list of ping locations for different Android phones
```
Comment (ghost) [inline android/src/com/mobiperf/measurements/TracerouteTask.java]: are you sure these locations work for all phones?

File: android/src/com/mobiperf/util/PhoneUtils.java [hunk 8]
```diff
@@ -695,23 +703,76 @@ private String intToIp(int number) {
     }
   }
   
+  @Deprecated
```
Comment (ghost) [inline android/src/com/mobiperf/util/PhoneUtils.java]: if this is deprecated, what should be used instead?

**Comment by ghost** [comment]
Added notes. I won't merge until they're dealt with.

it might be easier to switch over to the code review site that we've been using until now. I'm not sure why you haven't used it in this case.

File: android/src/com/mobiperf/measurements/TracerouteTask.java (hunk 7 above)
Comment (teamwork523) [inline android/src/com/mobiperf/measurements/TracerouteTask.java]: Right now we only use "/system/bin/ping", and I believe most of the program located in "/system/bin". I am not sure if Wenjie want to add more locations or not.

**Comment by ghost** [comment]
Code review complete: http://speedometer-reviews.appspot.com/48001/

**Comment by teamwork523** [comment]
Add ip type compatibility functionality in PhoneUtils.java
